src/main/java/teammates/ui/webapi/FeedbackSessionLogsUpdateAction.java

```diff
@@ -0,0 +1,94 @@
+import teammates.common.util.Logger;
+
+/**
+ * Action: Sync feedback session logs from GCloud logging service.
+ */
+public class FeedbackSessionLogsUpdateAction extends AdminOnlyAction {
+
+    private static final Logger log = Logger.getLogger();
+    private static final int MIN_WINDOW_PERIOD = 2 * 1000;
+    private static final int MAX_WINDOW_SIZE = 4;
+
+    @Override
+    public ActionResult execute() {
+        Instant currentTime = Instant.now();
+        Map<String, Map<String, List<FeedbackSessionLogEntryAttributes>>> studentToLogs =
+                new HashMap<>();
+        List<FeedbackSessionLogEntryAttributes> validLogEntries = new ArrayList<>();
+        List<FeedbackSessionLogEntryAttributes> allLogEntries =
+                logsProcessor.getFeedbackSessionLogs(null, null,
+                        currentTime.minus(15, ChronoUnit.MINUTES).toEpochMilli(), currentTime.toEpochMilli(), null);
+
+        // Arrange logs based on student email and log type.
+        for (FeedbackSessionLogEntryAttributes logEntry : allLogEntries) {
+            String studentEmail = logEntry.getStudentEmail();
+            String logType = logEntry.getFeedbackSessionLogType();
+            Map<String, List<FeedbackSessionLogEntryAttributes>> studentLog =
+                    studentToLogs.getOrDefault(studentEmail, new HashMap<>());
+
+            List<FeedbackSessionLogEntryAttributes> currList;
+
+            if (studentLog.containsKey(logType)) {
+                currList = studentLog.get(logType);
+            } else {
+                currList = new ArrayList<>();
+            }
+
+            currList.add(logEntry);
+            studentLog.put(logType, currList);
+            studentToLogs.put(studentEmail, studentLog);
+        }
+
+        for (Map.Entry<String, Map<String, List<FeedbackSessionLogEntryAttributes>>>
+                studentToLog : studentToLogs.entrySet()) {
+            Map<String, List<FeedbackSessionLogEntryAttributes>> studentLog =
+                    studentToLog.getValue();
+
+            int windowStartIndex = 0;
+            int windowSize = 0;
+            for (Map.Entry<String, List<FeedbackSessionLogEntryAttributes>>
+                    logEntries : studentLog.entrySet()) {
+                List<FeedbackSessionLogEntryAttributes> logs = logEntries.getValue();
+
+                for (int i = 0; i < logs.size(); i++) {
+                    FeedbackSessionLogEntryAttributes startLog = logs.get(windowStartIndex);
+                    FeedbackSessionLogEntryAttributes currLog = logs.get(i);
+
+                    if (currLog.getTimestamp() - startLog.getTimestamp() <= MIN_WINDOW_PERIOD) {
+                        windowSize++;
+                        // If the window size exceeds the max value
+                        // we only take the first log of the window.
+                        if (windowSize == MAX_WINDOW_SIZE) {
+                            startLog.setRemarks("This window has " + windowSize + " log(s)");
```

Comment:
This `remarks` field has some weird purpose.
- Why must it be a proper sentence? What is the downstream usage of this field?
  - In fact, why is it called `remarks`?
- Why not just record the number as is?

Reply:
`remarks` is to record abnormalities, and in this case, they are spams. We could just record the number, but I thought that might be confusing

Reply:
`remarks` is far more confusing. No part of the field says it's for abnormality, and it already assumes too much of downstream use case.
Whereas if you record the number, there can be many usages, because without context it is just a number.

And of course, the field name should not be `remarks`. If it represents number of accesses, the field name should say it clearly.